docs/csc-source-resolution.md

```diff
@@ -0,0 +1,14 @@
+# CatalogSourceConfig Source Resolution
```

Comment:
@galletti94 @kevinrizza @aravindhp please review this document that capture source resolution

Reply:
Please link to this doc from the main README

Reply:
This looks good to me. I'm going to wait to lgtm until aravindh has a chance to look over this latest change.